pkg/prometheus/operator.go

```diff
@@ -1274,6 +1274,24 @@ func (c *Operator) loadBasicAuthSecrets(
 				}
 				secrets[fmt.Sprintf("serviceMonitor/%s/%s/%d", mon.Namespace, mon.Name, i)] = credentials
 			}
+
+			if ep.BearerTokenSecret != nil {
+				sClient := c.kclient.CoreV1().Secrets(mon.Namespace)
+				token, err := getCredFromSecret(
+					sClient,
+					*ep.BearerTokenSecret,
+					"bearertoken",
+					mon.Namespace+"/"+ep.BearerTokenSecret.Name,
+					nsSecretCache,
+				)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to extract endpoint bearertoken for servicemonitor %v from secret %v in namespace %v",
+						mon.Name, ep.BearerTokenSecret.Name, mon.Namespace,
+					)
+				}
+				secrets[fmt.Sprintf("serviceMonitor/%s/%s/%d", mon.Namespace, mon.Name, i)] = BasicAuthCredentials{username: "", password: token}
```

Comment:
reusing BasicAuthCredentials here seems a little odd, maybe having a second map here would be more suitable?

Reply:
Right. I am moving it into a separate function to differentiate between basic auth and bearer tokens.